main.go

```diff
@@ -341,6 +350,18 @@ func (cmd *mainCmd) Run(args []string) error {
 	return multierr.Combine(errors...)
 }
 
+func checkGeneratedCode(f *ast.File) bool {
+	if ast.IsGenerated(f) {
+		return true
+	}
+	for _, comm := range f.Doc.List {
+		if strings.Contains(comm.Text, "@generated") {
```

Comment:
Should this be HasPrefix instead?

Reply:
Resolved offline - there are numerous cases where this isn't the case.

Reply:
There exists cases in our codebase where @generated is not the the Prefix.
Eg:
```
// Code generated by some tool. DO NOT EDIT. @generated
```